.github/workflows/build.yml

```
@@ -0,0 +1,22 @@
+name: build
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    paths:
+      - 'examples/*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
```

Here are some suggestions to:

* improve build times by caching go modules (even though for now it should be pretty quick.)
* print the used go version for debugging purposes

```suggestion
jobs:
  check:
    runs-on: ubuntu-latest

    steps:
      - name: Check out source code
        uses: actions/checkout@v2

      - name: Install Go
        uses: actions/setup-go@v2
        with:
          go-version: 1.16

      - name: Cache Go modules
        uses: actions/cache@v2
        with:
          path: ~/go/pkg/mod
          key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
          restore-keys: |
            ${{ runner.os }}-go-

      # Here, we simply print the exact go version, to have it as part of the
      # action's output, which might be convenient.
      - name: Print Go version
        run: go version
```

---

I like it! Caching is a great idea to save some resources.